src/schematools/loaders.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Union
+
+from more_ds.network.url import URL
+
+if TYPE_CHECKING:
+    from schematools.types import DatasetSchema
+
+
+class SchemaLoader:
+    """Base class for schema loaders."""
+
+    def __init__(self, schema_url: Union[Path, URL, str]):
+        """Initialize the schema loader.
+
+        schema_url:
+            Either a web url to the dataset schemas, or a path on the local filesystem.
+        """
+        self.schema_url = schema_url
```

Comment:
```python
self.schema_url = schema_url if isinstance(schema_url, Path) else URL(schema_url)
```

Daarna is het altijd een `Path` of een `URL` (en `URL` kan overal gebruikt worden waar een `str` wordt gebruikt.)